### PR TITLE
chore: program stage select refactor

### PR DESCRIPTION
**Pull request by turban**
This PR will refactor the ProgramStageSelect component to use: 
- Functional component
- DHIS2 app runtime (useDataQuery)
- Load data in component (skip redux)

First time I use `useDataQuery` and open for feedback!

### CODE REVIEW

**Review by jenniferarnesen — APPROVED** [review]

File: src/components/program/ProgramStageSelect.js [hunk 1]
```diff
@@ -1,83 +1,85 @@
...
-            loadProgramStages,
-            onChange,
-        } = this.props
+// Load all stages for one program
+const PROGRAM_STAGES_QUERY = {
+    stages: {
+        resource: 'programs',
+        id: ({ id }) => id,
+        params: {
+            fields: ['programStages[id,displayName~rename(name)]'],
```
Comment (jenniferarnesen) [inline src/components/program/ProgramStageSelect.js:13]: does it work if it is not an array, but a string? I'm not sure what is required.

File: src/components/program/ProgramStageSelect.js [hunk 2]
```diff
@@ -1,83 +1,85 @@
...
-            programStage,
-            programStages,
-            onChange,
-            className,
-            errorText,
-        } = this.props
-
-        if (!program) {
-            return null
+    // https://runtime.dhis2.nu/#/hooks/useDataQuery
```
Comment (jenniferarnesen) [inline src/components/program/ProgramStageSelect.js:38]: remove comment

File: src/components/program/ProgramStageSelect.js [hunk 3]
```diff
@@ -1,83 +1,85 @@
...
+            onComplete: onProgramStagesLoad,
+            lazy: true,
         }
+    )
 
-        let items = programStages[program.id]
+    // Fetch program stages when program is changed
+    useEffect(() => {
+        refetch({ id: program.id })
+    }, [program, refetch])
```
Comment (jenniferarnesen) [inline src/components/program/ProgramStageSelect.js:51]: technically you could use `program.id` in the dependency array, but it probably doesn't matter.

File: src/components/program/ProgramStageSelect.js (hunk 1 above)
Comment (turban) [reply src/components/program/ProgramStageSelect.js]: Switched to a string

File: src/components/program/ProgramStageSelect.js (hunk 3 above)
Comment (turban) [reply src/components/program/ProgramStageSelect.js:50]: Keeping program, but added program.id to propTypes

**Pull request merged**